src/Catalog/Helpers/Db2CatalogProjection.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Data;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    /// <summary>
+    /// Utility class to project <see cref="IDataRecord"/>s retrieved by db2catalog
+    /// into a format that is consumable by the catalog writer.
+    /// </summary>
+    public class Db2CatalogProjection
+    {
+        private readonly PackageContentUriBuilder _packageContentUriBuilder;
+
+        public Db2CatalogProjection(PackageContentUriBuilder packageContentUriBuilder)
+        {
+            _packageContentUriBuilder = packageContentUriBuilder ?? throw new ArgumentNullException(nameof(packageContentUriBuilder));
+        }
+
+        public FeedPackageDetails FromDataRecord(IDataRecord dataRecord)
+        {
+            if (dataRecord == null)
+            {
+                throw new ArgumentNullException(nameof(dataRecord));
+            }
+
+            var packageId = dataRecord["Id"].ToString();
+            var normalizedPackageVersion = dataRecord["NormalizedVersion"].ToString();
+            var listed = dataRecord.GetBoolean(dataRecord.GetOrdinal("Listed"));
```

Comment:
Do the IDataRecord.GetBoolean throw if the conversion fails or return a default value? If silently fails is there any way to ensure the expected type?

Reply:
Depends on the concrete implementation. In the case of `SqlDataReader`, an `InvalidCastException` would be thrown if the data type doesn't match (no conversions happen): https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqldatareader.getboolean?view=netframework-4.8